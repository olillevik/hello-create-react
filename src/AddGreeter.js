import React, {Component} from 'react';
import './AddGreeter.css';

class AddGreeter extends Component {

    constructor(props) {
        super(props);
        this.state = {greetingName: ''}
        this.handleUpdate = this.handleUpdate.bind(this);
        this.addGreeting = this.addGreeting.bind(this);
    }

    render() {
        return (
            <div className='AddGreeter'>
                <input type='text' onChange={this.handleUpdate}/>
                &nbsp;&nbsp;
                <button onClick={this.addGreeting}>add</button>
            </div>
        );
    }

    handleUpdate(event) {
        this.setState({greetingName: event.target.value});
    }

    addGreeting() {
        this.props.addGreeting(this.state.greetingName);
        this.setState({greetingName: ''});
    }
}

export default AddGreeter;