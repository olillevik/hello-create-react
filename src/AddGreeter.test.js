import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import AddGreeter from './AddGreeter';

describe(AddGreeter, () => {
    const mockAddGreeting = jest.fn();
    const component = shallow(
        <AddGreeter addGreeting={mockAddGreeting}/>
    );

    it('renders and matches our snapshot', () => {
        const component1 = renderer.create(
            <AddGreeter/>
        );
        const tree = component1.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the form', () => {
        expect(component.find('input')).toHaveLength(1);
        expect(component.find('button')).toHaveLength(1);
    });

    it('calls passed in function when add button is clicked', () => {
        component.find('button').simulate('click');
        expect(mockAddGreeting).toBeCalled();
    });

    it('updates the form when keys are pressed', () => {
        const updateKey = 'foo';
        component.instance().handleUpdate({
            target: {
                value: updateKey
            }
        });

        expect(component.state('greetingName')).toEqual(updateKey);
    });

    it('blanks out the form when the button is clicked', () => {
        const updateKey = 'bar';
        component.instance().handleUpdate({
            target: {
                value: updateKey
            }
        });
        component.find('button').simulate('click');
        expect(component.state('greetingName')).toBe('');
    });
});

