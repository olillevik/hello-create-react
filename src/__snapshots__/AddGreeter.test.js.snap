// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class AddGreeter extends _react.Component {

    constructor(props) {
        super(props);
        this.state = { greetingName: '' };
        this.handleUpdate = this.handleUpdate.bind(this);
        this.addGreeting = this.addGreeting.bind(this);
    }

    render() {
        return (
            _react2.default.createElement('div', { className: 'AddGreeter', __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: this },
                _react2.default.createElement('input', { type: 'text', onChange: this.handleUpdate, __source: { fileName: _jsxFileName, lineNumber: 16 }, __self: this }), '\\xA0\\xA0',

                _react2.default.createElement('button', { onClick: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 18 }, __self: this }, 'add')));


    }

    handleUpdate(event) {
        this.setState({ greetingName: event.target.value });
    }

    addGreeting() {
        this.props.addGreeting(this.state.greetingName);
        this.setState({ greetingName: '' });
    }} renders and matches our snapshot 1`] = `
<div
  className="AddGreeter"
>
  <input
    onChange={[Function]}
    type="text"
  />
    
  <button
    onClick={[Function]}
  >
    add
  </button>
</div>
`;
